src/main/java/software/aws/neptune/gremlin/GremlinConnectionProperties.java

```diff
@@ -1095,12 +1098,23 @@ public void setLoadBalancingStrategy(@NonNull final LoadBalancingStrategy strate
     protected void validateProperties() throws SQLException {
         // If IAMSigV4 is specified, we need the region provided to us.
         if (getAuthScheme() != null && getAuthScheme().equals(AuthScheme.IAMSigV4)) {
-            final String region = System.getenv().get("SERVICE_REGION");
-            if (region == null) {
-                throw new SQLException(
-                        "Error: SERVICE_REGION environment variable must be set to use IAM authentication. " +
-                                "Note, it must be accessible from BI tools if using a BI tool. " +
-                                "Example SERVICE_REGION=us-east-1.");
+            if ("".equals(getRegion())) {
+                if (System.getenv("SERVICE_REGION") == null) {
+                    throw missingConnectionPropertyError(
+                            "A Service Region must be provided to use IAMSigV4 Authentication through " +
+                                    "the SERVICE_REGION environment variable or the serviceRegion connection property. " +
+                                    "If you are using the driver with a BI Tools, you will need to set the serviceRegion " +
+                                    "property to the appropriate value. For example, 'serviceRegion=us-east-1'.");
+
+                }
+                // Technically this doesn't need to be set if we are using "SERVICE_REGION" in Gremlin
+                setRegion(System.getenv("SERVICE_REGION"));
+            } else {
+                try {
+                    setRegionEnv(getRegion());
```

Comment:
Would it be more logical if the region provided in the connection string would always override the environment variable?

Reply:
Yes, any non-empty string values will override the existing environment variable. Unless I've missed something in my logic?